R/arrange.R

```diff
@@ -68,22 +64,40 @@ arrange <- function(.data, ..., .by_group = FALSE) {
 
 #' @param .by_group If `TRUE`, will sort first by grouping variable. Applies to
 #'   grouped data frames only.
+#' @param .locale The locale to sort character vectors in.
+#'
+#'   - Defaults to [dplyr_locale()], which uses the American English locale
+#'     if the stringi package is installed and the global default locale
+#'     has not been altered. See the help page for [dplyr_locale()] for the
+#'     exact details.
+#'
+#'   - If a single string is supplied, then this will be used as the locale
+#'     identifier to sort with. For example, `"fr"` will sort with the French
+#'     locale. This requires the stringi package. Use
+#'     [stringi::stri_locale_list()] to generate a list of possible locale
+#'     identifiers.
```

Comment:
```suggestion
#'   - If a single string is supplied, then this will be used as the sorting locale
#'     For example, `"fr"` will sort with the French locale. This requires the stringi 
#'     package. Use [stringi::stri_locale_list()] to generate a list of possible locale
#'     identifiers.
```

Is there a precise name for these sorts of identifiers?

Reply:
I don't think so. The ICU documentation calls them "locale IDs" but doesn't get more specific than that when referring to the whole locale string, so that is what I went off of.

For `"en_US"`, it refers to the `"en"` part as a _language code_, and the `"US"` part as a _country code_

https://unicode-org.github.io/icu/userguide/locale/#the-locale-concept